qsimcirq/qsim_simulator.py

```diff
@@ -160,8 +164,12 @@ def _sample_measure_results(
             ValueError: If there are multiple MeasurementGates with the same key,
                 or if repetitions is negative.
         """
-        if not isinstance(program, qsimc.QSimCircuit):
-            program = qsimc.QSimCircuit(program, device=program.device)
+
+        # Add noise to the circuit if a noise model was provided.
+        program = qsimc.QSimCircuit(
+            self.noise.noisy_moments(program, program.all_qubits()),
```

Comment:
Why are qubits `sorted()` in the base implementation but not here?

Reply:
Good call, thanks!